### PR TITLE
fix: avoid to encode twice the State param when response_mode is form…

### PR DESCRIPTION
**Pull request by leleueri**
…_post

related to AM-2744

### CODE REVIEW

**Review by lgw-gravitee — APPROVED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class AuthorizationResponseTest {
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java:34]: Remove this 'public' modifier.

Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java:34]: Remove this 'public' modifier.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    private final static String STATE = "state+value==";
+
+    @ParameterizedTest
+    @MethodSource("responseProvider")
+    public void should_manage_state_encoding(AuthorizationResponse response) {
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java:40]: Remove this 'public' modifier.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        assertEquals(UriBuilder.encodeURIComponent(STATE), response.params().get(Parameters.STATE));
+        assertEquals(UriBuilder.encodeURIComponent(STATE), response.params(true).get(Parameters.STATE));
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponseTest.java:42]: Call to 'assertEquals()' from 'Assert' should be replaced with a call to a method from 'org.junit.jupiter.api.Assertions'

**Review by mstasgravitee — APPROVED** [review]

**Pull request merged**